csrc/host_ir/host_ir.cpp

```diff
@@ -71,7 +83,9 @@ NVFUSER_DEFINE_CLONE_AND_CREATE(PostOnStream)
 std::string PostOnStream::toString(int indent_size) const {
   int indent_increment = 2;
   std::stringstream ss;
-  indent(ss, indent_size) << "Execute the following Fusion, taking inputs :{\n";
+  indent(ss, indent_size) << "PostOnStream the operation :{\n";
+  indent(ss, indent_size) << hostOpToPost()->toString();
```

Comment:
```suggestion
  indent(ss, indent_size) << hostOpToPost();
```

IIRC, we've defined `operator<<` for Expr* so it's the same as calling toString. But please double check that.

Reply:
Right! I'll fix that

Reply:
Actually, that overload didn't seem to work consistently and nobody couldn't figure out why. I remember it didn't work for my build (just raw addressed displayed) but worked for someone else. We ended up switching to be more explicit with `toString()`.

Reply:
ok let me reverse that then, ok?

> Actually, that overload didn't seem to work consistently and nobody couldn't figure out why.

naive guess: maybe because `ir/iostream.h` header is missing sometimes?

Reply:
> naive guess: maybe because ir/iostream.h header is missing sometimes?

Very likely. https://github.com/NVIDIA/Fuser/blob/f0bbb74a97bfca6c4810473513c6b094391a7be7/csrc/ir/iostream.h#L93

That's why it's important to IWYU, which however lacks a good tool to autocomplete and enforce. For example, when seeing `operator<<(const Statement*)`, a tool wouldn't know whether the intention is to print the address or the content. 

That being said, `operator<<(const Statement&)` may work better because people will have to include the right header to compile. However, the downside is that the input can't be null. 

> ok let me reverse that then, ok?

I prefer including `iostream.h` and saving a bit of typing here and there. But either way SGTM.